### PR TITLE
Use the ledger's Coin instead of our custom Lovelace type

### PR DESCRIPTION
**Pull request by smelc**
# Changelog

```yaml
- description: |
    Use the ledger's Coin instead of our custom Lovelace type
# uncomment types applicable to the change:
  type:
  # - feature        # introduces a new feature
  - breaking       # the API has changed in a breaking way
  # - compatible     # the API has changed but is non-breaking
  # - optimisation   # measurable performance improvements
  - improvement    # QoL changes e.g. refactoring
  # - bugfix         # fixes a defect
  # - test           # fixes/modifies tests
  # - maintenance    # not directly related to the code
  # - release        # related to a new release preparation
  # - documentation  # change in code docs, haddocks...
```

# Context

* Reboot of https://github.com/IntersectMBO/cardano-api/pull/368
* Fixes https://github.com/IntersectMBO/cardano-api/issues/473

# Checklist

- [X] Commit sequence broadly makes sense and commits have useful messages
- [X] Self-reviewed the diff

### CODE REVIEW

**Review by carbolymer — APPROVED** [review]

File: cardano-api/test/cardano-api-golden/files/golden/errors/Cardano.Api.Fees.TxBodyErrorAutoBalance/TxBodyErrorAdaBalanceNegative.txt [hunk 1]
```diff
@@ -1 +1 @@
-The transaction does not balance in its use of ada. The net balance of the transaction is negative: Lovelace 1 lovelace. The usual solution is to provide more inputs, or inputs with more ada.
\ No newline at end of file
+The transaction does not balance in its use of ada. The net balance of the transaction is negative: Coin 1 lovelace. The usual solution is to provide more inputs, or inputs with more ada.
```
[the remark this answers is not in the document]
Comment (smelc) [reply cardano-api/test/cardano-api-golden/files/golden/errors/Cardano.Api.Fees.TxBodyErrorAutoBalance/TxBodyErrorAdaBalanceNegative.txt]: Done in the two additional commits :+1:

File: cardano-api/internal/Cardano/Api/Orphans.hs [hunk 2]
```diff
@@ -189,6 +190,8 @@ deriving instance Data Bech32.CharPosition
 deriving newtype instance Real L.Coin
 deriving newtype instance Integral L.Coin
 deriving newtype instance Num L.Coin
+instance Pretty L.Coin where
+  pretty (L.Coin n) = pretty n <+> "lovelace"
```
Comment (carbolymer) [inline cardano-api/internal/Cardano/Api/Orphans.hs:194]:
```suggestion
  pretty (L.Coin n) = pretty n <+> "Lovelace"
```
Lovelace is capitalized.

Comment (smelc) [reply cardano-api/internal/Cardano/Api/Orphans.hs]: Good point, changed :+1:

File: cardano-api/internal/Cardano/Api/Value.hs [hunk 3]
```diff
@@ -47,15 +47,11 @@ module Cardano.Api.Value
     -- * Internal conversion functions
   , toByronLovelace
   , fromByronLovelace
-  , toShelleyLovelace
```
Comment (Jimbo4350) [inline cardano-api/internal/Cardano/Api/Value.hs:50]: :+1:

**Review by Jimbo4350 — COMMENTED** [review]
:+1:  We should have probably deprecated the `Lovelace` type with a warning and then removed it fully after a release.

**Pull request merged**